### PR TITLE
MINOR: Should commit a task if the consumer position advanced as well

### PR DESCRIPTION
**Pull request by guozhangwang**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by guozhangwang** [comment]
@vvcephei @AlanConfluent this is not a ready to merge PR, but just demonstrating what I meant to fix the issue.

I will add the test coverage after you've agreed this is the right fix.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java [hunk 1]
```diff
@@ -1158,7 +1158,38 @@ public String toString(final String indent) {
...
+            return true;
+        } else {
+            for (final Map.Entry<TopicPartition, Long> entry : consumedOffsets.entrySet()) {
+                final TopicPartition partition = entry.getKey();
+                try {
+                    final long offset = mainConsumer.position(partition);
+
+                    // note the position in consumer is the "next" record to fetch,
+                    // so it should be larger than the consumed offset by 1; if it is
+                    // more than 1 it means there are skipped offsets
```
Comment (vvcephei) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java]: "skipped offsets" sounds scary. How about "control records, which the consumer skips over silently"?

**Review by vvcephei — COMMENTED** [review]
Thanks @guozhangwang !

Yes, this looks like the right fix to me.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java (hunk 1 above)
Comment (guozhangwang) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java]: ack

**Review by vvcephei — APPROVED** [review]
Thanks, @guozhangwang !

**Pull request merged**